src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -844,20 +854,35 @@ private void ExecuteBlockActionsCore<TBlockStartAction, TBlockAction, TNodeActio
             // Execute stateful executable node analyzers, if any.
             if (executableNodeActions.Any())
             {
+                var nodesToAnalyzeBuilder = ArrayBuilder<TNode>.GetInstance();
+                addNodesToAnalyze(executableBlocks, nodesToAnalyzeBuilder);
+
                 if (syntaxNodeActions != null)
                 {
                     Debug.Assert(getKind != null);
 
+                    var syntaxNodesToAnalyzeBuilder = ArrayBuilder<SyntaxNode>.GetInstance(nodesToAnalyzeBuilder.Count);
+                    foreach (var syntaxNode in nodesToAnalyzeBuilder)
+                        syntaxNodesToAnalyzeBuilder.Add((syntaxNode as SyntaxNode)!);
```

Comment:
Why `(syntax as SyntaxNode)!` and not (SyntaxNode)syntaxNode? If this is not a `SyntaxNode` shouldn't we find out here vs. null refing later?

Reply:
Direct cast to SyntaxNode or IOperation doesn't compile. I will go ahead and change it to cast to object and then to SyntaxNode or IOperation assuming that is preferable. 

Btw, it's not completely obvious but the calling code either puts in IOperation or SyntaxNodes, and doesn't intermingle.